### ISSUE TITLE
Missing new line breaks in output

### ISSUE THREAD

**Issue by sushidub** [mrodrig/json-2-csv-cli#4]
#### version
`@mrodrig/json-2-csv-cli@3.7.3`

 #### json chunk
```
[
  {
      "link": {
          "@href": "http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/1"
      },
      "latitude": "31.0069",
      "longitude": "-88.0103",
      "name": "Barry"
  }, {
      "link": {
          "@href": "http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/2"
      },
      "latitude": "34.0136",
      "longitude": "-85.9703",
      "name": "Gadsden"
  }, {
      "link": {
          "@href": "http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/3"
      },
      "latitude": "33.6446",
      "longitude": "-87.2003",
      "name": "Gorgas"
  }, {
      "link": {
          "@href": "http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/4"
      },
      "latitude": "32.6017",
      "longitude": "-87.7811",
      "name": "Greene County"
  }, {
     .....
```
---
#### json2csv current output
*note the string 'undefined' after the headers*
```sh
 $ json2csv Desktop/facilities.json
link.@href,latitude,longitude,nameundefinedhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/1,31.0069,-88.0103,Barry,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/2,34.0136,-85.9703,Gadsden,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/3,33.6446,-87.2003,Gorgas,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/4,32.6017,-87.7811,Greene County,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/5,33.2442,-86.4567,E C Gaston,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/6,34.7439,-87.8486,Colbert,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/7,34.8825,-85.7547,Widows Creek,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/8,31.4833,-87.9125,Charles R Lowman,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/9,31.4001,-86.4767,McWilliams,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/10,33.6319,-87.0597,James H Miller Jr,http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/11,31.2542,-88.0319,McIntosh (7063),http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/12,31.2561,-88.0103,Washington County Cogen (Olin)
```
*subsequently the spreadsheet import summary reflects the same problem*
<img width="846" alt="json-2-csv-import-current" src="https://user-images.githubusercontent.com/482622/91937476-3a506280-ecaf-11ea-8e00-3684b3f79c11.png">

---
Same issue with --eol flags

#### current [with --eol flag]
*note the string literal interpretation of the new line character anywhere we'd expect new line*

```sh
$ json2csv Desktop/facilities.json --eol "\n"
link.@href,latitude,longitude,name\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/1,31.0069,-88.0103,Barry\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/2,34.0136,-85.9703,Gadsden\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/3,33.6446,-87.2003,Gorgas\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/4,32.6017,-87.7811,Greene County\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/5,33.2442,-86.4567,E C Gaston\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/6,34.7439,-87.8486,Colbert\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/7,34.8825,-85.7547,Widows Creek\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/8,31.4833,-87.9125,Charles R Lowman\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/9,31.4001,-86.4767,McWilliams\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/10,33.6319,-87.0597,James H Miller Jr\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/11,31.2542,-88.0319,McIntosh (7063)\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/12,31.2561,-88.0103,Washington County Cogen (Olin)
```
```sh
 $ json2csv Desktop/facilities.json --eol "\\n"
link.@href,latitude,longitude,name\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/1,31.0069,-88.0103,Barry\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/2,34.0136,-85.9703,Gadsden\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/3,33.6446,-87.2003,Gorgas\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/4,32.6017,-87.7811,Greene County\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/5,33.2442,-86.4567,E C Gaston\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/6,34.7439,-87.8486,Colbert\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/7,34.8825,-85.7547,Widows Creek\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/8,31.4833,-87.9125,Charles R Lowman\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/9,31.4001,-86.4767,McWilliams\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/10,33.6319,-87.0597,James H Miller Jr\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/11,31.2542,-88.0319,McIntosh (7063)\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/12,31.2561,-88.0103,Washington County Cogen (Olin)
```
```sh
 $ json2csv Desktop/facilities.json --eol \n
link.@href,latitude,longitude,namenhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/1,31.0069,-88.0103,Barrynhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/2,34.0136,-85.9703,Gadsdennhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/3,33.6446,-87.2003,Gorgasnhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/4,32.6017,-87.7811,Greene Countynhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/5,33.2442,-86.4567,E C Gastonnhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/6,34.7439,-87.8486,Colbertnhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/7,34.8825,-85.7547,Widows Creeknhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/8,31.4833,-87.9125,Charles R Lowmannhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/9,31.4001,-86.4767,McWilliamsnhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/10,33.6319,-87.0597,James H Miller Jrnhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/11,31.2542,-88.0319,McIntosh (7063)nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/12,31.2561,-88.0103,Washington County Cogen (Olin)
```
```sh
 $ json2csv Desktop/facilities.json --eol \\n
link.@href,latitude,longitude,name\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/1,31.0069,-88.0103,Barry\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/2,34.0136,-85.9703,Gadsden\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/3,33.6446,-87.2003,Gorgas\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/4,32.6017,-87.7811,Greene County\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/5,33.2442,-86.4567,E C Gaston\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/6,34.7439,-87.8486,Colbert\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/7,34.8825,-85.7547,Widows Creek\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/8,31.4833,-87.9125,Charles R Lowman\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/9,31.4001,-86.4767,McWilliams\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/10,33.6319,-87.0597,James H Miller Jr\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/11,31.2542,-88.0319,McIntosh (7063)\nhttp://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/12,31.2561,-88.0103,Washington County Cogen (Olin)
```
---

#### expected output

```sh
 $ json2csv Desktop/facilities.json
link.@href,latitude,longitude,name
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/1,31.0069,-88.0103,Barry
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/2,34.0136,-85.9703,Gadsden
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/3,33.6446,-87.2003,Gorgas
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/4,32.6017,-87.7811,Greene County
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/5,33.2442,-86.4567,E C Gaston
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/6,34.7439,-87.8486,Colbert
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/7,34.8825,-85.7547,Widows Creek
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/8,31.4833,-87.9125,Charles R Lowman
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/9,31.4001,-86.4767,McWilliams
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/10,33.6319,-87.0597,James H Miller Jr
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/11,31.2542,-88.0319,McIntosh (7063)
http://10.1.6.16:8080/emit/resources/nations/1/states/1/facilities/12,31.2561,-88.0103,Washington County Cogen (Olin)
```
*and equally expected is the spreadsheet import summary based on good output*
<img width="798" alt="json-2-csv-import-patch" src="https://user-images.githubusercontent.com/482622/91937784-f1e57480-ecaf-11ea-8efd-1671476de3f8.png">

---

#### my temporary patch

*json2csv.js*

```diff
220c220
<         }).join(options.delimiter.eol);
---
>         }).join(options.delimiter.eol ? options.delimiter.eol : constants.defaultOptions.delimiter.eol);
351c351
<                 (options.prependHeader ? header + options.delimiter.eol : '') +
---
>                 (options.prependHeader ? header + (options.delimiter.eol ? options.delimiter.eol : constants.defaultOptions.delimiter.eol) : '') +
```

**Comment by mrodrig** [comment]
Thanks for reporting this @sushidub and sorry for the delay in getting back to you. I took a look at this and found there's a bug in the logic that's specifically building the delimiter portion of the options object. I'm going to work on getting a PR out to fix that in the https://github.com/mrodrig/json-2-csv repo and should have that published later today. I'll publish a new release of this CLI that's updated to use that version of the `json-2-csv` package as well so that it will pull in the fixed version by default.

**Comment by mrodrig** [comment]
Thanks again. This should be fixed in `3.7.13`. Please let me know if you're still seeing this issue.
